pkgs/development/python-modules/web3/default.nix

```diff
@@ -42,12 +43,13 @@ buildPythonPackage rec {
     eth-typing
     eth-utils
     hexbytes
-    ipfshttpclient
     jsonschema
     lru-dict
     protobuf
     requests
     websockets
+  ] ++ lib.optionals (!stdenv.isDarwin) [
+    ipfshttpclient # ipfshttpclient is broken on darwin
```

Comment:
It would be better to put it in `passthru.optional-dependencies` rather than `propagatedBuildInputs`.
However, it will not be implicitly included in the package. @RaitoBezarius  san, what do you think?
https://github.com/ethereum/web3.py/blob/a90b963a1f09b3f2417754c5f186ae451f6fd20b/setup.py#L43-L45

Reply:
I just read what's https://ryantm.github.io/nixpkgs/stdenv/stdenv/#var-stdenv-passthru about, it seems quite fitting. Since the ipfshttpclient is more like a runtime plugin, instead of a build-time dependency.

Reply:
I don't quite know how to write this one properly. I'd like some input from the maintainer @RaitoBezarius for sure.

Reply:
This section should be helpful.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#how-to-add-optional-dependencies-python-optional-dependencies

Reply:
Apologies for the delay in response, I removed myself as a maintainer of web3 as I do not have further time to spare on it.
From my memory, the IPFS stuff was quite broken when I last time checked, I would not bother too much on it until you have a need for it.

Reply:
@RaitoBezarius no problems. I will see what I can do with this one.

Reply:
@natsukium updated using passthru.